### ISSUE TITLE
two pop-ups generated in google docs

### ISSUE THREAD

**Issue by monzug** [alpheios-project/alpheios-core#289]
related to issue #169 
1) comment from March 23:
I do get the message that mouse move event is enabled. however the lookup works funny. I did enter a greek word e.g. βουλεύω and initially it did generate the morphology pop-up but then it went back to the lexical query produced no results or it does open two pop-up: one for lookup and the other for the word I did enter in lookup. have you seen this?
2) comment and screenshot from April 29:
when I use alpheios on google doc (mouseover or double click) I get two pop-ups. I did try with Chinese words and greek words and I got two pop-ups in both scenarios.
see screenshots
![chinese-doc](https://user-images.githubusercontent.com/41396793/80721935-d1709800-8acc-11ea-9d91-6ba67d9f1b26.JPG)
![two-pop-ups](https://user-images.githubusercontent.com/41396793/80721968-dc2b2d00-8acc-11ea-9382-0d44018a07fa.JPG)

**Comment by monzug** [comment]
here, one of my google doc sample
μάλιστα δὲ πρὸς τὴν φύσιν· ἔστι γὰρ οἷον ἀρχὴ
τῶν ζῴων. Ἐπιζητοῦμεν δὲ θεωρῆσαι καὶ γνῶναι τήν τε
φύσιν αὐτῆς καὶ τὴν οὐσίαν, εἶθ’ ὅσα συμβέβηκε περὶ αὐτήν
「學而時習之，不亦說乎？有朋自遠方來，不亦樂乎？人不知而不慍，不亦君子乎？」

**Comment by monzug** [comment]
Ok, I can see two pop-ups been generated after a loooong lookup but one is visible only for a short time, not enough to take a screenshot

**Comment by irina060981** [comment]
I tried to reproduce on google docs with Monica's text sample, but was not able.
All my lookups were vere fast :)

**Comment by monzug** [comment]
Irina, can you try to enter something really loooooong in lookup? That's how I got the two pop-ups yesterday but this was not the same scenario as original reported.

**Comment by monzug** [comment]
just saw it again in Chrome on PC. I have a docs with greek and chinese, and page language is syriac. when I mousemove on any greek word I get two pop-ups: lexical query fails for syriac and fails for latin.

**Comment by balmas** [comment]
hmm. I still can't reproduce it, even when I set the mousemove delay really low (10ms). 

In theory it shouldn't be possible as we only ever add one popup element to the page. Not sure what is going on here.

**Comment by monzug** [comment]
again today
![Screen Shot 2020-05-28 at 3 11 06 PM](https://user-images.githubusercontent.com/41396793/83145766-9e361e80-a0f5-11ea-9043-eb8362195465.png)

**Comment by balmas** [comment]
fascinating. @monzug does it happen only on the first lookup after loading the page?

**Comment by monzug** [comment]
yes, but it's double clicking I think that's causing it. I get one pop-up for latin and one pop-up for greek (which is the page language, I just changed it)

**Comment by monzug** [comment]
nope, all lookups

**Comment by balmas** [comment]
> yes, but it's double clicking I think that's causing it. I get one pop-up for latin and one pop-up for greek (which is the page language, I just changed it)

One thing to double check - if mousemove is enabled or not. Sometimes I have found myself clicking when mousemove is enabled and it is difficult to tell that it's actually responding to the mousemovement and not the double-click

**Comment by monzug** [comment]
I didn’t check that but I will. Isn’t double click the default option now?

**Comment by balmas** [comment]
Yes it should be, but if you previously had mousemove selected it might not have cleared that.

**Comment by monzug** [comment]
I closed google docs and alpheios before going to the dentist. as soon as I got back, I opened the google doc again and activate Alpheios, checked the mousemove option and confirm that it was not checked. Anyhow doubleclicking works fine now. so it could be the influence of the testing that I was doing on the other tabs, before opening google docs????

**Comment by balmas** [comment]
>so it could be the influence of the testing that I was doing on the other tabs, before opening google docs????

that would be really weird but I guess not impossible.

**Comment by monzug** [comment]
I do not know what to think. I opened google docs, double clicked on few
words and got the two pop-ups. so first I thought it had to do with page
language so I changed it from latin to greek, but I kept getting the two
pop-ups. then I deactivated alpheios and closed google docs. on reopening,
I was not longer able to repro it.

**Comment by balmas** [comment]
inspecting the contents of a Google Docs page in the debugger, I'm wondering if it's possible that in some scenarios the alpheios popup is getting appended to a document in an iframe rather than the main document, and that's how the two popups are appearing at the same time. In theory that should not be possible but the Google Docs source is filled with hidden iframes.  The other idea I had is that maybe Google Docs is doing something with Shadow DOM elements that is confusing our code. (see discussion started in #397 )

**Comment by monzug** [comment]
Why am I the only one experiencing this? my testing environment is different than yours, for sure. in more than 2 months I am still the only one seeing the two pop-ups. it could be an extremely edgy case (a combination of page language with who knows what), in which case I don't know if it's worth spending any time on it. what do you think?

**Comment by balmas** [comment]
I agree it's probably an edge case, but the fact that it clearly can happen when the code in theory should make it not possible indicates that there is something we need to at least pay attention to.

I agree we should not hold up the release for it however.

**Comment by monzug** [comment]
absolutely. I will spend more time testing it during regression.

**Comment by monzug** [comment]
this time I open google docs, and activate alpheios. double clicking was not working so I check in Options for mousemove. it was checked. I click on reset all. then first double click I got the two pop-ups

![Screen Shot 2020-05-29 at 2 49 40 PM](https://user-images.githubusercontent.com/41396793/83261621-eb7cc380-a1bb-11ea-9294-5c2a0e2595c5.png)

**Comment by monzug** [comment]
![Screen Shot 2020-05-29 at 2 52 49 PM](https://user-images.githubusercontent.com/41396793/83261761-2a127e00-a1bc-11ea-956a-65b666f3de70.png)

**Comment by balmas** [comment]
@monzug can you give me the link to the doc you are testing with?

There is a problem with mousemove activation in the latest build (see #400)  but I want to see if there is any way I can reproduce using the exact same document you are.

**Comment by monzug** [comment]
μάλιστα δὲ πρὸς τὴν φύσιν· ἔστι γὰρ οἷον ἀρχὴ

τῶν ζῴων. Ἐπιζητοῦμεν δὲ θεωρῆσαι καὶ γνῶναι τήν τε

φύσιν αὐτῆς καὶ τὴν οὐσίαν, εἶθ’ ὅσα συμβέβηκε περὶ αὐτήν


「學而時習之，不亦說乎？有朋自遠方來，不亦樂乎？人不知而不慍，不亦君子乎？」

**Comment by balmas** [comment]
no, not the contents :-) the actual document

**Comment by monzug** [comment]
https://docs.google.com/document/d/1J-YPkuvy1jSz7REE0YKetyL3dDXcSmxjYWNbkJnoXiQ/edit

**Comment by monzug** [comment]
uhm, I just got the two pop-ups on latin library. so, I change env-webext.js to point to dev database, then I reinstall the build after clearing data. activate alpheios on latin library, double click on a word and got the two pop-ups, I immediately went to check the mouseover in Options, but it's unchecked. see screeshot

![Screen Shot 2020-05-29 at 4 20 11 PM](https://user-images.githubusercontent.com/41396793/83270519-e07c6000-a1c8-11ea-85d6-d4d9754ef53a.png)
![Screen Shot 2020-05-29 at 4 19 16 PM](https://user-images.githubusercontent.com/41396793/83270526-e2deba00-a1c8-11ea-80ca-51111660bd51.png)

**Comment by balmas** [comment]
@monzug is there any chance you have 2 versions of the alpheios extensions installed??

What do you see in chrome://extensions?

**Comment by monzug** [comment]
to retest in regression
